hal/targets/hal/TARGET_ARM_SSG/TARGET_BEETLE/sleep.c

```diff
@@ -0,0 +1,30 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sleep_api.h"
+#include "cmsis.h"
+#include "system_core_beetle.h"
+
+ void sleep(void)
+ {
+    SystemPowerSuspend(POWER_MODE_SLEEP);
```

Comment:
That depends on the use case, you need to guarantee it'll be woken up if you put it to sleep. My next step is to implement tickless sleep and the wake up source will be lp_ticker